ejb3/src/main/java/org/jboss/as/ejb3/subsystem/EJB3SubsystemDefaultPoolWriteHandler.java

```diff
@@ -27,40 +27,84 @@
+    private static final String DEFAULT_MDB_POOL_CONFIG_CAPABILITY_NAME = "org.wildfly.ejb3.pool-config.mdb-default";
+    private static final String DEFAULT_ENTITY_POOL_CONFIG_CAPABILITY_NAME = "org.wildfly.ejb3.pool-config.entity-default";
 
     public static final EJB3SubsystemDefaultPoolWriteHandler SLSB_POOL =
-            new EJB3SubsystemDefaultPoolWriteHandler(StrictMaxPoolConfigService.DEFAULT_SLSB_POOL_CONFIG_SERVICE_NAME,
-                    EJB3SubsystemRootResourceDefinition.DEFAULT_SLSB_INSTANCE_POOL);
+            new EJB3SubsystemDefaultPoolWriteHandler(DEFAULT_SLSB_POOL_CONFIG_CAPABILITY_NAME, EJB3SubsystemRootResourceDefinition.DEFAULT_SLSB_INSTANCE_POOL);
+
+    public static final EJB3SubsystemDefaultPoolWriteHandler MDB_POOL =
+            new EJB3SubsystemDefaultPoolWriteHandler(DEFAULT_MDB_POOL_CONFIG_CAPABILITY_NAME, EJB3SubsystemRootResourceDefinition.DEFAULT_MDB_INSTANCE_POOL);
 
     public static final EJB3SubsystemDefaultPoolWriteHandler ENTITY_BEAN_POOL =
-            new EJB3SubsystemDefaultPoolWriteHandler(StrictMaxPoolConfigService.DEFAULT_ENTITY_POOL_CONFIG_SERVICE_NAME,
-                    EJB3SubsystemRootResourceDefinition.DEFAULT_ENTITY_BEAN_INSTANCE_POOL);
+            new EJB3SubsystemDefaultPoolWriteHandler(DEFAULT_ENTITY_POOL_CONFIG_CAPABILITY_NAME, EJB3SubsystemRootResourceDefinition.DEFAULT_ENTITY_BEAN_INSTANCE_POOL);
 
-    private final ServiceName poolConfigServiceName;
+    private final String poolConfigCapabilityName;
     private final AttributeDefinition poolAttribute;
 
-    public EJB3SubsystemDefaultPoolWriteHandler(ServiceName poolConfigServiceName, AttributeDefinition poolAttribute) {
+    public EJB3SubsystemDefaultPoolWriteHandler(String defaultPoolConfigCapabilityName, AttributeDefinition poolAttribute) {
         super(poolAttribute);
-        this.poolConfigServiceName = poolConfigServiceName;
+        this.poolConfigCapabilityName = defaultPoolConfigCapabilityName;
         this.poolAttribute = poolAttribute;
     }
 
+    /*
+     * Update the conditional capabilities for the default bean instance pools if the attribute values have changed
+     * This write handler is registered with the EJB3SubsystemRootResource
+     */
+    @Override
+    protected void recordCapabilitiesAndRequirements(OperationContext context, AttributeDefinition attributeDefinition, ModelNode newValue, ModelNode oldValue) {
+        Resource resource = context.readResource(PathAddress.EMPTY_ADDRESS);
+        ModelNode model = resource.getModel();
+
+        // NOTE: the old value does not contain an expression, but the new value contains an expression, so we need to resolve it first
+        ModelNode resolvedNewValue = new ModelNode();
+        try {
+            resolvedNewValue = context.resolveExpressions(newValue);
+        } catch(OperationFailedException ofe) {
+            // TODO: handle any possible exception correctly
```

Comment:
Probably log a WARN and just deregister the old requirement, don't record the new one.

Reply:
Done